File: backend-csharp/DTOs/Responses/TokenResponse.cs
namespace Picture2Text.Api.DTOs.Responses;

/// <summary>
/// Token 回應資料
/// </summary>
public class TokenData
{
    /// <summary>
    /// Access Token（用於 API 訪問）
    /// </summary>
    public string AccessToken { get; set; } = string.Empty;

    /// <summary>
    /// Refresh Token（用於刷新 Access Token）
    /// </summary>
    public string RefreshToken { get; set; } = string.Empty;

    /// <summary>
    /// Token 類型
    /// </summary>
    public string TokenType { get; set; } = "Bearer";

    /// <summary>
    /// Access Token 過期時間
    /// </summary>
    public DateTime ExpiresAt { get; set; }

    /// <summary>
    /// Refresh Token 過期時間
    /// </summary>
    public DateTime RefreshTokenExpiresAt { get; set; }

    /// <summary>
    /// 會話 ID
    /// </summary>
    public string SessionId { get; set; } = string.Empty;
}

/// <summary>
/// Token 回應
/// </summary>
public class TokenResponse : ApiResponse<TokenData>
{
}

File: backend-nestjs/src/dto/response/token.response.ts
import { ApiProperty } from '@nestjs/swagger';

export class TokenData {
  @ApiProperty({
    description: 'Access Token (JWT)',
    example: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...',
  })
  AccessToken: string;

  @ApiProperty({
    description: 'Refresh Token',
    example: 'dGhpcyBpcyBhIHJlZnJlc2ggdG9rZW4...',
  })
  RefreshToken: string;

  @ApiProperty({
    description: 'Token 類型',
    example: 'Bearer',
  })
  TokenType: string;

  @ApiProperty({
    description: 'Access Token 過期時間',
    example: '2024-01-15T10:30:00.000Z',
  })
  ExpiresAt: Date;

  @ApiProperty({
    description: 'Refresh Token 過期時間',
    example: '2024-01-22T10:00:00.000Z',
  })
  RefreshTokenExpiresAt: Date;

  @ApiProperty({
    description: 'Session ID',
    example: '550e8400-e29b-41d4-a716-446655440000',
  })
  SessionId: string;
}

export class TokenResponse {
  @ApiProperty()
  Code: number;

  @ApiProperty()
  Message: string;

  @ApiProperty({ type: TokenData })
  Data: TokenData;
}
